src/pke/lib/schemebase/base-pke.cpp

```diff
@@ -201,8 +202,25 @@ std::shared_ptr<std::vector<Element>> PKEBase<Element>::EncryptZeroCore(
                   ? Element(dgg, elementParams, Format::EVALUATION)
                   : Element(tug, elementParams, Format::EVALUATION);
 
-  Element e0(dgg, elementParams, Format::EVALUATION);
-  Element e1(dgg, elementParams, Format::EVALUATION);
+  //Element e0(dgg, elementParams, Format::EVALUATION);
+  //Element e1(dgg, elementParams, Format::EVALUATION);
+
+  Element e0;
+  Element e1;
+
+  auto preMode = cryptoParams->GetPREMode();
+  std::cout << "base-pke encryptzerocore premode set " << preMode << std::endl;
+  std::cout << "base-pke encryptzerocore noise distribution parameter " << cryptoParams->GetFloodingDistributionParameter();
+
+  //if ((preMode == FIXED_NOISE_HRA) || (preMode == NOISE_FLOODING_HRA)) {
+  if (dgg.GetStd() == 1) {
```

Comment:
as we have discussed, you should have created a function checking the state of dgg. dgg.GetStd() is wrong

Reply:
As discussed, you were going to add a IsValid function to dgg because it is applicable outside of PRE too. I can add it if you want me to instead.

Reply:
call this function isInitialized() and compare std and 1 using abs().